rust_grpcio_bench/src/main.rs

```diff
@@ -0,0 +1,58 @@
+mod proto;
+
+use std::io::Read;
+use std::sync::Arc;
+use std::{io, thread};
+
+use futures::channel::oneshot;
+use futures::executor::block_on;
+use futures::prelude::*;
+use grpcio::{ChannelBuilder, Environment, RpcContext, ServerBuilder, UnarySink};
+
+use crate::proto::gen::helloworld::{HelloReply, HelloRequest};
+use crate::proto::gen::helloworld_grpc::{create_greeter, Greeter};
+
+#[derive(Clone)]
+struct GreeterService;
+
+impl Greeter for GreeterService {
+    fn say_hello(&mut self, ctx: RpcContext<'_>, req: HelloRequest, sink: UnarySink<HelloReply>) {
+        let msg = format!("{}", req.get_name());
+        let mut resp = HelloReply::default();
+        resp.set_message(msg);
```

Comment:
wouldn't it be faster to just pass `req.get_name().clone()` directly?

Reply:
maybe
but it's cleaner anyway

Reply:
How about `let msg = req.get_name().clone()`? My point is, I'm not sure this format macro is going to nicely optimize itself to just clone the underlying string. Or will it?

Reply:
Yeah that's what I did 🚀